internal/oci/runtime_oci.go

```diff
@@ -638,19 +644,23 @@ func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout in
 		}
 	}
 
-	return waitContainerStop(ctx, c, killContainerTimeout, false)
+	err = waitContainerStop(ctx, c, killContainerTimeout, false)
+	if err == nil {
+		c.SetFinished(time.Now())
+		return nil
+	}
+	return err
 }
 
 func checkProcessGone(c *Container) error {
-	process, perr := findprocess.FindProcess(c.state.Pid)
+	process, perr := findprocess.FindProcess(c.Pid())
```

Comment:
Same thing here. get pid in the beginning instead of multiple calls to c.Pid()

Reply:
Addressed